### PR TITLE
Fixes #28920 - Add queries and mutation for settings

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: app/graphql/foreman_graphql_schema.rb [hunk 1]
```diff
@@ -6,11 +6,13 @@ class ForemanGraphqlSchema < GraphQL::Schema
   query(Types::Query)
   mutation(Types::Mutation)
 
-  rescue_from ActiveRecord::RecordInvalid, &:message
-  rescue_from ActiveRecord::Rollback, &:message
-  rescue_from StandardError, &:message
-  rescue_from ActiveRecord::RecordNotUnique, &:message
-  rescue_from ActiveRecord::RecordNotFound, &:message
+  if Rails.env.production?
```
Comment (xprazak2) [inline app/graphql/foreman_graphql_schema.rb:9]: To show stack trace in server logs when in development.

File: app/graphql/mutations/base_mutation.rb [hunk 2]
```diff
@@ -30,6 +30,20 @@ def attribute_required?(attribute)
 
     def authorize!(resource, action)
       user = context[:current_user]
+      if resource.class.included_modules.include?(Authorizable)
```
Comment (xprazak2) [inline app/graphql/mutations/base_mutation.rb]: settings do not have permissions, only admin can change them

Comment (tbrisker) [reply app/graphql/mutations/base_mutation.rb]: There are actually `view_settings` and `edit_settings` permissions, although it looks like they were not fully implemented - they are only used to limit access to the endpoints and in the ui

Comment (xprazak2) [reply app/graphql/mutations/base_mutation.rb]: Right, that was for canned admin... lets see how many things will break when I try to make settings authorizable...

File: test/models/setting_test.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 require 'test_helper'
+require 'pry'
```
Comment (timogoebel) [inline test/models/setting_test.rb]: Nit: Don't forget to remove this.

Comment (xprazak2) [reply test/models/setting_test.rb]: Removed.

**Review by timogoebel — DISMISSED** [review]
Looks good to me.

File: app/graphql/mutations/settings/update.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module Mutations
+  module Settings
+    class Update < UpdateMutation
+      graphql_name 'UpdateSettingMutation'
+      description 'Updates a setting'
+
+      argument :id, ID, required: true
```
Comment (shiramax) [inline app/graphql/mutations/settings/update.rb]: id shouldn't be part of the general update?

File: app/graphql/mutations/settings/update.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+      argument :value, String
```
Comment (shiramax) [inline app/graphql/mutations/settings/update.rb]: The value should also be required.

File: test/graphql/queries/settings_query_test.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      GRAPHQL
+    end
+
+    let(:data) { result['data']['settings'] }
+
+    test 'fetch settings' do
+      assert_empty result['errors']
+
+      expected_count = Setting.count
+      assert_not_equal 0, expected_count
```
Comment (shiramax) [inline test/graphql/queries/settings_query_test.rb:34]: can you add also the tests from here: 
https://github.com/theforeman/foreman/blob/develop/test/controllers/api/v2/settings_controller_test.rb

File: test/models/setting_test.rb [hunk 7]
```diff
@@ -667,11 +667,21 @@ def sticky_setting
...
+    settings = Setting.live_descendants.select(&:full_name)
+    a_foo = settings.index { |item| item.name == a_name }
+    b_foo = settings.index { |item| item.name == b_name }
+    c_foo = settings.index { |item| item.name == c_name }
+    assert a_foo
+    assert b_foo
+    assert c_foo
+    assert a_foo < b_foo
+    assert b_foo < c_foo
   end
```
Comment (shiramax) [inline test/models/setting_test.rb:685]: can you explain why this was changed?

**Review by shiramax — CHANGES_REQUESTED** [review]

File: app/graphql/mutations/settings/update.rb (hunk 4 above)
Comment (kamils-iRonin) [reply app/graphql/mutations/settings/update.rb]: `id` is already defined in `UpdateMutation`
https://github.com/theforeman/foreman/blob/3caf38a0d7dc28773c79394dca6d4816e7144ee9/app/graphql/mutations/update_mutation.rb#L5

File: test/models/setting_test.rb (hunk 7 above)
Comment (xprazak2) [reply test/models/setting_test.rb:685]: Long story short:

```ruby
settings = Setting.live_descendants.map(&:full_name).compact
#=> ["Access unattended without build", "A Foo Name", "B Foo Name", "C Foo Name"]
settings.sort
#=> ["A Foo Name", "Access unattended without build", "B Foo Name", "C Foo Name"]
```

By adding proper `setting_type` to one of the [fixtures](https://github.com/theforeman/foreman/pull/7411/files#diff-44b5b50a5d1889681cb7411ca971905a) I exposed a behaviour that made the test fail. Apparently, using order on postgres gives [surprising results](https://stackoverflow.com/questions/34537982/incorrect-sort-collation-order-with-spaces-in-postgresql-9-4) when whitespace is involved.

**Comment by xprazak2** [comment]
[test foreman]

**Comment by xprazak2** [comment]
[test katello]

**Comment by shiramax** [comment]
test failures are related.

**Comment by shiramax** [comment]
test failures are related.

**Comment by xprazak2** [comment]
Finally all green.

File: app/graphql/mutations/settings/update.rb (hunk 4 above)
Comment (timogoebel) [reply app/graphql/mutations/settings/update.rb]:
```suggestion
```
@kamils-iRonin: This line is redundant, correct?

Comment (kamils-iRonin) [reply app/graphql/mutations/settings/update.rb]: I think so

Comment (xprazak2) [reply app/graphql/mutations/settings/update.rb]: I missed that, thanks.

**Review by shiramax — APPROVED** [review]
thanks @xprazak2

**Pull request merged**